parser_library/include/debugger.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2019 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+#ifndef HLASMPLUGIN_PARSERLIBRARY_DEBUGGER_H
+#define HLASMPLUGIN_PARSERLIBRARY_DEBUGGER_H
+
+#include "parser_library_export.h"
+#include "protocol.h"
+#include "range.h"
+#include "sequence.h"
+#include "workspace_manager.h"
+
+namespace hlasm_plugin::parser_library::debugging {
+
+// Interface that a listener can implement to be notified about debugging events.
+class debug_event_consumer
+{
+protected:
+    ~debug_event_consumer() = default;
+
+public:
+    virtual void stopped(sequence<char> reason, sequence<char> addtl_info) = 0;
+    virtual void exited(int exit_code) = 0;
+
+    void stopped(std::string_view reason, std::string_view addtl_info)
+    {
+        stopped(sequence(reason), sequence(addtl_info));
+    }
+};
+
+class breakpoints_t
```

Comment:
why can't breakpoints_t be implemented using sequence<breakpoint>?

Reply:
It could be, now that the sequence can optionally own data, but I would basically just return sequence<breakpoint, breakpoints_t>, which seemed overcomplicated given that breakpoint is the public type and is stored in vector.